File: codeforces/practice/453A/pony.cpp
#include <bits/stdc++.h>

using namespace std;

double power(double a, int n) {
  double ans = 1.0;
  while (n > 0) {
    if (n % 2 != 0) {
       ans = a * ans;
    }
    a = a * a;
    n = n / 2;
  }
  return ans;
}

int main() {
  int n, m;
  double ans = 0.0;
  cin >> m >> n;
  for (int i=1; i<=m; i++) {
    ans = ans + (i*(power((1.0*i)/m, n) - power(((1.0*i)-1)/m, n)));
  }
  printf("%f\n", ans);
  return 0;
}

File: codeforces/practice/453A/pony.scala
import scala.io.StdIn._

object Main {
  
  def power(a: Double, n: Int): Double = {
    var ans = 1.0
    var b = a
    var p = n
    while (p > 0) {
      if ( (p & 0x01) != 0) ans = b * ans
      b = b * b
      p = p >> 1;
    }
    ans
  }
  def main(args: Array[String]): Unit = {
    val mn = readLine.split(" ").map(_.toInt)
    val m = mn.head
    val n = mn.last
    val ans = (1 to m).map { k =>
      k * (power((k*1.0)/m, n) - power(((k*1.0) - 1.0)/m, n))
    }.sum
    println(ans)
  }
}
